lti_consumer/lti13.py

```diff
@@ -0,0 +1,187 @@
+from django.conf import settings
+
+from urllib.parse import urlencode
+
+from Crypto.PublicKey import RSA
+from jwkest.jwk import RSAKey
+from jwkest.jws import JWS
+import json
+from jwkest import jwk
+
+
+LTI_BASE_MESSAGE = {
+    # Claim type: fixed key with value `LtiResourceLinkRequest`
+    # http://www.imsglobal.org/spec/lti/v1p3/#message-type-claim
+    "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiResourceLinkRequest",
+
+    # LTI Claim version
+    # http://www.imsglobal.org/spec/lti/v1p3/#lti-version-claim
+    "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
+
+    # Optional claims - useless ones
+    # "https://purl.imsglobal.org/spec/lti/claim/context": {
+    #     "id": "c1d887f0-a1a3-4bca-ae25-c375edcc131a",
+    #     "label": "ECON 1010",
+    #     "title": "Economics as a Social Science",
+    #     "type": ["http://purl.imsglobal.org/vocab/lis/v2/course#CourseOffering"]
+    # },
+    # "https://purl.imsglobal.org/spec/lti/claim/tool_platform": {
+    #     "guid": "ex/48bbb541-ce55-456e-8b7d-ebc59a38d435",
+    #     "contact_email": "support@platform.example.edu",
+    #     "description": "An Example Tool Platform",
+    #     "name": "Example Tool Platform",
+    #     "url": "https://platform.example.edu",
+    #     "product_family_code": "ExamplePlatformVendor-Product",
+    #     "version": "1.0"
+    # },
+
+    # Optional claims - Useful
+    # This is useful for error redirects
+    # http://www.imsglobal.org/spec/lti/v1p3/#launch-presentation-claim
+    "https://purl.imsglobal.org/spec/lti/claim/launch_presentation": {
+        "document_target": "iframe",  # iframe, frame, window
+        # Endpoint to redirect the user to after completing LTI task
+        # Returns with log statements
+        "return_url": "https://platform.example.edu/terms/201601/courses/7/sections/1/resources/2"
+    },
+    # Custom variables :)
+    "https://purl.imsglobal.org/spec/lti/claim/custom": {
+        "xstart": "2017-04-21T01:00:00Z",
+        "request_url": "https://tool.com/link/123"
+    }
+}
+
+
+class LtiConsumer1p3:
+    def __init__(
+            self,
+            iss,
+            lti_oidc_url,
+            lti_launch_url,
+            deployment_id,
+            rsa_key,
+    ):
+        self.iss = iss
+        self.oidc_url = lti_oidc_url
+        self.launch_url = lti_launch_url
+        self.deployment_id = deployment_id
+
+        # Generate JWK from RSA key
+        self.jwk = RSAKey(
```

Comment:
I'm storing the RSA key as plain text in an XBlock's text field.

We might need to change the implementation once we decide were the tool is going to be stored.